tests/unit/test_charm.py

```diff
@@ -224,6 +224,22 @@ def test_given_not_matching_csr_and_certificate_when_provide_certificate_action_
         self.harness.charm._on_provide_certificate_action(event=event)
         event.fail.assert_called_once_with(message="Certificate and CSR do not match.")
 
+    def test_given_invalid_ca_chain_when_provide_certificate_action_then_event_fails(self):
```

Comment:
Here we should mock the call to `ca_chain_is_valid`, we don't want to validate this public method's behavior both here and in the helpers tests.